sdk/typescript/src/providers/provider.ts

```diff
@@ -140,6 +145,97 @@ export abstract class Provider {
 
   abstract syncAccountState(address: string): Promise<any>;
 
+  /**
+   * Get events for one transaction
+   * @param digest transaction digest to search by
+   * @param count max result count
+   */
+  abstract getEventsByTransaction(digest: TransactionDigest, count: number): Promise<SuiEvents>;
+
+  /**
+   * Get events emitted from within the specified Move module
+   * @param package_ Move package object ID
+   * @param module Move module name
+   * @param count max result count
+   * @param startTime start of time range
+   * @param endTime end of time range
+   */
+  abstract getEventsByModule(
+    package_: ObjectId,                   // 'package' is reserved word
```

Comment:
```suggestion
    packageId: ObjectId, 
```
You can get rid of this comment by using `packageId`